memphis/partition_generator.py

```diff
@@ -0,0 +1,12 @@
+
+
+class PartitionGenerator:
+    def __init__(self, partitions_list):
+        self.partitions_list = partitions_list
+        self.current = 0
+        self.length = len(partitions_list)
+
+    def next(self):
```

Comment:
If you rename this to `__next__`, you can then use the more Pythonic approach of working with iterators: `next(iter)`